WordPress/gutenberg
# Autocomplete: Check key events more strictly in IME composition

Follow-up on #45496

I would like to see this PR backported to a minor release in addition to #45496.
Sorry for the additional correspondence.

## What?
This PR adds logic in the Autocomplete component to strictly check whether an IME composition session is in progress.

## Why?

An IME composition issue was also discovered in the FormTokenField component and will be fixed in #45607. However, as this comment suggests, there are some cases that cannot be controlled depending on the combination of OS and browser.


## How?
Added event.isComposing to determine if an IME composition session is in progress.

## Testing Instructions

This PR requires Japanese input, but can be tested by following the steps below:

- Change the site language to Japanese.
- Open the Submission Editor.
- In an empty paragraph tag, enter the text `/見出し` which is the Japanese translation of the heading block.
- Confirm that the list of candidates is narrowed down when you press the `enter` key.
- Confirm that the heading block is inserted as expected when you press the `enter` key again.
